reframe/core/buildsystems.py

```diff
@@ -782,6 +792,94 @@ def generated_modules(self):
         return self._eb_modules
 
 
+class Spack(BuildSystem):
+    '''A build system for building test code using `Spack
+    <https://spack.io/>`__.
+
+    ReFrame will use Spack to build and install the code given an environment
+    and a set of specs.
+
+    .. versionadded:: 3.6.1
+
+    '''
+
+    #: The list of specs to build and install within the given environment.
+    #:
+    #: ReFrame will add the specs to the active environment by emititing the
+    #: following command:
+    #:
+    #: .. code-block:: bash
+    #:
+    #:    spack add spec1 spec2 spec3
+    #:
+    #: :type: :class:`List[str]`
+    #: :default: ``[]``
+    specs = fields.TypedField(typ.List[str])
+
+    #: Spack environment.
+    #: This field is required.
+    #:
+    #: ReFrame activates this environment to build and install the specified
+    #: packages:
+    #:
+    #: .. code-block:: bash
+    #:
+    #:    spack env activate -d <environment directory>
+    #:
+    #: :type: :class:`str` or :class:`None`
+    #: :default: :class:`None`
+    environment = fields.TypedField(str, type(None))
+
+    def __init__(self):
+        super().__init__()
+        self.specs = []
+        self.environment = None
+        self._prefix_save = None
+
+    def emit_build_commands(self, environ):
+        self._prefix_save = os.getcwd()
+        if not self.environment:
+            raise BuildSystemError(f"'environment' must not be empty")
+
+        ret = self._env_activate_cmds()
+        if self.specs:
+            specs_str = ' '.join(self.specs)
+            ret.append(f'spack add {specs_str}')
+
+        ret.append('spack install')
+        return ret
+
+    def _env_activate_cmds(self):
+        return [f'. $SPACK_ROOT/share/spack/setup-env.sh',
+                f'spack env activate -d {self.environment}']
```

Comment:
@rsarm, in order to avoid issues related to view creation, I would add the flag `--without-view` this command line.
More info can be found here https://spack.readthedocs.io/en/latest/environments.html